### DISCUSSION TITLE
RAGFlow Server setup for all users.

### DISCUSSION THREAD

**Discussion by mdre** [infiniflow/ragflow#3565] [Ideas]
We need to share a specialized knowledge database with all our users but we can't find a way to setup a global server. 
We work at the Prosecutor Offices in our state and want to have a database with all laws added there so we can chat with the AI about a criminal case. 
To setup the server as we need, there's some feature that we can't find en RAGFlow:
1. An admin user profile: this user will setup permisions, create databases that could be shared across all the server and configure all parameters.
2. Server must have only one default config, so everybody can use the same parameters. For example, we want to setup RAGFlow integrated with Ollama. Thats must be only one config for all conections.
3. User that access the server only have a Chat option to start a thread about some case file and use the default database.
4. User can bind a on-the-fly database to chat about a specific case data with the IA but that data should be local to that chat. Perhaps it could be handy to have a ID of that database if we want to share it in differents chats channels with all others investigators of the case. 
5. The API should let to create this channels to start a conversation and should be a way to preserve the chat session for future comments. 

RAGFlow it really good. We are testing another's but seems RF will be a good choice if we can find a way to do what we need.
Great Job!!

**Comment by wouldbe12** [comment]
Have you managed to adapt RAGFlow for this? I would also need something similar. Many thanks,

**Reply by mdre** [reply]
Sorry but no for now.  The project will be addressed  in 2025 and we hope to find a way to do what we need.

**Comment by PentaFrame-Development** [comment]
Good morning!
In my opinion, what you could do at the moment is:
- Setup a single 'admin' profile where you create the llm connections and datasets. 
- Create an agent with the begin block that contains a dataset variable. 
- Embed the agent in a webpage so everyone can use it.

With this structure you can select the dataset when the agent starts and chat with it like in the chat but with different datasets everytime. 
I'll attach a screenshot to explain better.

<img width="2294" height="1127" alt="image" src="https://github.com/user-attachments/assets/ac2a47c8-db60-4ecb-8aff-635f391be991" />
<img width="2330" height="1207" alt="image" src="https://github.com/user-attachments/assets/1b284387-3ccd-4286-8cab-ffc08dc57f0a" />
<img width="2324" height="804" alt="image" src="https://github.com/user-attachments/assets/1a00ba9d-70cf-4056-b341-4081048cfbb2" />

With this structure, adding the agent that you need next to the retrieval you can bind knowledge base 'on the fly' and if you need it for multiuser, just share the dataset with team and they will be able to use the dataset too, but the model configuration is needed for every profile for what i know. 

Hope you'll find my suggestion helpful and feel free to ask for any info or question! 
Thanks.
PentaFrame-Development
